dnsdist: Add the ability to dump a summary of the cache content

rgacogne opened the pull request:
### Short description
<!-- Write a small description of what this Pull Request fixes or provides, including the issue #s -->
The new feature writes a summary of the packet cache content to a file, dumping for each entry:
- qname
- remaining TTL
- qtype
-  computed hash
- response's length
- TCP or UDP
- the timestamp of when the entry was added to the cache. 

### Checklist
<!-- please indicate if any of these things are done/included with this Pull Request. Not all boxes need to be checked for the Pull Request to be accepted -->
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [x] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

phonedph1 commented:
This is much better than what I had come up with!

pieterlexis commented:
Looks good. Perhaps the TTL and QType should be swapped?

Also, I would like a 'header' on the first line like this:

```
; qname TTL type
```

rgacogne commented:
>  Perhaps the TTL and QType should be swapped?

I had the same feeling but decided to keep the same order than the one when dumping the recursor's packet cache. But the more I think about it, the less relevant it is so I'm OK with swapping it.

> Also, I would like a 'header' on the first line like this:

I'm not sure it's really needed, because that's almost what we already have:
```
; dnsdist's packet cache dump follows
;
powerdns.com. 3572 A ; key 2000355501, length 57, tcp 0, added 1529912626
www.powerdns.com. 3590 A ; key 4247463626, length 61, tcp 0, added 1529912644
nx.powerdns.com. 3595 A ; key 1069543761, length 109, tcp 0, added 1529912649
```
